services-core/src/gas_price/gnosis_safe.rs

```diff
@@ -59,34 +58,30 @@ const SAFE_LOW_PERCENTILE: f64 = 0.3;
 // price can be 1000 times the fast gas price which is not reasonable.
 const SECONDS_PER_BLOCK: f64 = 15.0;
 // Treat percentiles as probabilities for geometric distribution.
-const FAST_TIME: f64 = SECONDS_PER_BLOCK * FAST_PERCENTILE;
-const STANDARD_TIME: f64 = SECONDS_PER_BLOCK * STANDARD_PERCENTILE;
-const SAFE_LOW_TIME: f64 = SECONDS_PER_BLOCK * SAFE_LOW_PERCENTILE;
+const FAST_TIME: f64 = SECONDS_PER_BLOCK / FAST_PERCENTILE;
+const STANDARD_TIME: f64 = SECONDS_PER_BLOCK / STANDARD_PERCENTILE;
+const SAFE_LOW_TIME: f64 = SECONDS_PER_BLOCK / SAFE_LOW_PERCENTILE;
 
 /// Retrieve gas prices from the Gnosis Safe gas station service.
 #[derive(Debug)]
-pub struct GnosisSafeGasStation {
-    client: HttpClient,
-    uri: Uri,
+pub struct GnosisSafeGasStation<T> {
+    transport: T,
+    uri: &'static str,
 }
 
-impl GnosisSafeGasStation {
-    pub fn new(http_factory: &HttpFactory, api_uri: &str) -> Result<GnosisSafeGasStation> {
-        let client = http_factory.create()?;
-        let uri: Uri = api_uri.parse()?;
-        Ok(GnosisSafeGasStation { client, uri })
+impl<T: Transport> GnosisSafeGasStation<T> {
+    pub fn new(transport: T, uri: &'static str) -> GnosisSafeGasStation<T> {
+        GnosisSafeGasStation { transport, uri }
     }
 
     /// Retrieves the current gas prices from the gas station.
     pub async fn gas_prices(&self) -> Result<GasPrices> {
-        self.client
-            .get_json_async(&self.uri, HttpLabel::GasStation)
-            .await
+        self.transport.get_json(&self.uri).await
     }
 }
 
 #[async_trait::async_trait]
-impl GasPriceEstimating for GnosisSafeGasStation {
+impl<T: Transport + Send + Sync> GasPriceEstimating for GnosisSafeGasStation<T> {
```

Comment:
Maybe, we can just make the Transport `Send + Sync` so we only need `Transport` trait bound?